admin/class-yoast-notification-center.php

```diff
@@ -130,50 +221,333 @@ public function set_transient() {
+	 *
+	 * @param string $key Key to retrieve.
+	 *
+	 * @return mixed value of key if set.
+	 */
+	private static function get_user_input( $key ) {
+		$filter_input_type = INPUT_GET;
+		if ( 'POST' === filter_input( INPUT_SERVER, 'REQUEST_METHOD' ) ) {
+			$filter_input_type = INPUT_POST;
+		}
+
+		return filter_input( $filter_input_type, $key );
+	}
+
+	/**
+	 * Get Notification Identifier
+	 *
+	 * @param string|Yoast_Notification $notification Notification to get the ID of.
+	 *
+	 * @return bool|string Identifier if found, False if not found.
+	 */
+	private static function get_notification_id( $notification ) {
+		if ( $notification instanceof Yoast_Notification ) {
+			return $notification->get_id();
+		}
+
+		if ( is_string( $notification ) ) {
+			return $notification;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get the dismissal key for the notification
+	 *
+	 * @param string|Yoast_Notification $notification Notification to get the dismissal key of.
+	 *
+	 * @return bool|string Dismissal key if found, False if not found.
+	 */
+	private static function get_dismissal_key( $notification ) {
+		if ( $notification instanceof Yoast_Notification ) {
+			return $notification->get_dismissal_key();
+		}
+
+		if ( is_string( $notification ) ) {
+			return $notification;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Keep a list of conditions so we don't add duplicates
+	 *
+	 * @param Yoast_Notification_Condition $condition Condition to add to the stack.
+	 */
+	public function add_notification_condition( Yoast_Notification_Condition $condition ) {
+		// Prevent duplicates.
+		if ( $this->has_notification_condition( $condition ) ) {
+			return;
+		}
+
+		$this->notification_conditions[] = $condition;
+	}
+
+	/**
+	 * Check if the notification condition is already registered
+	 *
+	 * @param Yoast_Notification_Condition $condition Condition to check for.
+	 *
+	 * @return bool
+	 */
+	private function has_notification_condition( Yoast_Notification_Condition $condition ) {
+		return in_array( $condition, $this->notification_conditions, true );
+	}
+
+	/**
+	 * Check if the notification can be shown for the current user
+	 *
+	 * @param Yoast_Notification $notification Notification to check.
+	 *
+	 * @return bool
+	 */
+	private function show_notification( Yoast_Notification $notification ) {
+		// Don't display if it has been dismissed for the current user.
+		if ( $this->maybe_dismiss_notification( $notification ) ) {
+			return false;
+		}
+
+		// Don't display if the user doesn't have enough capabilities.
+		return $notification->display_for_current_user();
+	}
+
+	/**
+	 * Get the notifications from storage
+	 *
+	 * @return array Yoast_Notification[] Notifcations
+	 */
+	private function get_notifications_from_storage() {
+
+		// The notifications array.
+		$notifications = array();
+
+		$stored_notifications = get_option( self::STORAGE_KEY );
+
+		// Check if transient is set.
+		if ( false !== $stored_notifications ) {
```

Comment:
If you pass `array()` as a default to `get_option` you don't need this `if`-statement.